ui/desktop/electron-app/src/spawn-promise.js

```diff
@@ -45,22 +46,37 @@ module.exports = {
       childProcess.stdout.on('data', (data) => {
         outputStream += data.toString();
         const jsonData = jsonify(outputStream);
-        if (jsonData) resolve(jsonData);
+        if (jsonData) {
+          if (jsonData.termination_reason) {
+            spawnSession.cancelProcess(childProcess.pid);
+          } else {
+            // Track only successfully launched child processes
+            spawnSession.add({ childProcess, data: jsonData });
+          }
+          outputStream = '';
```

Comment:
Indeed. Because `outputStream` is scoped to the function, I think could be declared as `const` on line 47. So we do not need to concat values and clean it up at the end?

Reply:
Ah.  A promise may only be resolved or rejected once.  So it doesn't make sense to continue receiving input after the initial resolution.  Can the data event handler be removed after resolution?